engine/class_modules/sc_demon_hunter.cpp

```diff
@@ -2372,6 +2372,28 @@ struct fel_devastation_t : public demon_hunter_spell_t
     }
   }
 
+  void last_tick( dot_t* d ) override
+  {
+    demon_hunter_spell_t::last_tick( d );
+
+    if ( p()->talent.vengeance.darkglare_boon->ok() )
+    {
+      // CDR reduction and Fury refund are separate rolls per Realz
+      double base_cooldown = p()->talent.vengeance.fel_devastation->cooldown().total_seconds();
+      timespan_t minimum_cdr_reduction =
+          timespan_t::from_seconds( p()->talent.vengeance.darkglare_boon->effectN( 1 ).percent() * base_cooldown );
+      timespan_t maximum_cdr_reduction =
+          timespan_t::from_seconds( p()->talent.vengeance.darkglare_boon->effectN( 2 ).percent() * base_cooldown );
+      timespan_t cdr_reduction   = rng().range( minimum_cdr_reduction, maximum_cdr_reduction );
+      double minimum_fury_refund = p()->talent.vengeance.darkglare_boon->effectN( 3 ).base_value();
+      double maximum_fury_refund = p()->talent.vengeance.darkglare_boon->effectN( 4 ).base_value();
+      double fury_refund         = rng().range( minimum_fury_refund, maximum_fury_refund );
+
+      p()->cooldown.fel_devastation->adjust( cdr_reduction );
```

Comment:
This value needs to be passed in as -cdr_reduction since right now this would add to the cooldown.